File: demo/.jdbrc.py
import sys
def breakfunc(*args):
	id=args[0]
	threadName=args[1]
	
	expr=e('testStr')
	print "jython break:testStr="+ expr
	print "jython break:id="+ str(id)
	print "jython break:threadName="+ threadName
	if expr=="abc":
		_('list')
	else:
		_('cont')
	
addBreakPoint("XJdbTest:15",breakfunc, True)

def breakfunc(*args):
	id=args[0]
	
	expr=e('testStr')
	print "jython break:testStr="+ expr
	if expr=="abc":
		_('list')
		e('this.callA()')
		e('this.callB("script")')
		_p("-------------------------")
		_('p this.callA()')
		_('p this.callB("script")')
	else:
		_('cont')
	
addBreakPoint("XJdbTest:42",breakfunc, True)

def stopEvent(threadName):
	_p("[stop event] threadName is "+threadName)


addStopEvent(stopEvent)

def exitedEvent(cause):
	_p("exited event cause="+str(cause))

addExitedEvent(exitedEvent)

File: demo/.jdbrc.js
var bkstring = "XJdbTest:15";
var bp = addBreakPoint(bkstring,function(bp,id,threadName)
{
	var expr=$('testStr');
	_p('javascript break:expr='+expr);
	_p('javascript break:id='+id);
	_p('javascript break:threadName='+threadName);
			
	if(expr=="abc")
		_('list');
	else
		_('cont');	
},true);

var bkstring = "XJdbTest:42";
var bp = addBreakPoint(bkstring,function(bp,id)
{
	var expr=$('testStr');				
	if(expr=="abc")
	{
		_('list')
		$('this.callA()')
		$('this.callB("script****************")')
		_p("-------------------------")
		_('p this.callA()')
		_('p this.callB("script")')
	}
	else
		_('cont');	
},true);


function stop_event(threadName)
{
	_p("[stop event] threadName is "+threadName)	
}

addStopEvent(stop_event)

addExitedEvent(function(cause)
{
	_p("exited event cause="+cause)
});
